src/main/java/com/google/firebase/auth/FirebaseUserManager.java

```diff
@@ -230,20 +231,26 @@ UserImportResult importUsers(UserImportRequest request) throws FirebaseAuthExcep
 
   Tenant getTenant(String tenantId) throws FirebaseAuthException {
     GenericUrl url = new GenericUrl(tenantMgtBaseUrl + "/tenants/" + tenantId);
-    Tenant response = sendRequest("GET", url, null, Tenant.class);
-    if (Strings.isNullOrEmpty(response.getTenantId())) {
-      throw new FirebaseAuthException(TENANT_NOT_FOUND_ERROR, "Failed to get tenant.");
-    }
-    return response;
+    return sendRequest("GET", url, null, Tenant.class);
+  }
+
+  Tenant createTenant(Tenant.CreateRequest request) throws FirebaseAuthException {
+    GenericUrl url = new GenericUrl(tenantMgtBaseUrl + "/tenants");
+    return sendRequest("POST", url, request.getProperties(), Tenant.class);
+  }
+
+  Tenant updateTenant(Tenant.UpdateRequest request) throws FirebaseAuthException {
+    Map<String, Object> properties = request.getProperties();
+    checkArgument(!properties.isEmpty(), "tenant update must have at least one property set");
+    String updateMask = String.join(",", ImmutableSortedSet.copyOf(properties.keySet()));
```

Comment:
This will break if the request contains any nested properties (we don't have any at the moment). If the request is shaped like `{"foo": {"bar": 100}}`, the update mask should look like `['foo.bar']`.

Let's either implement a small util function to generate the correct update masks for all requests, or add a comment/TODO here to indicate the limitation with the current appraoch.

Reply:
Done.